### ISSUE TITLE
Provider produced inconsistent result after apply - additional_trust_bundle

### ISSUE THREAD

**Issue by am90670** [terraform-redhat/terraform-provider-rhcs#229]
When providing additional trusted certificates via the `proxy.additional_trust_bundle` attribute the provider attempts to store a different value in the state than what Terraform expects. Causing Terraform to exit with the error (I've replaced the certificate bodies with ...)

```shell
rhcs_cluster_rosa_classic.rosa_sts_cluster: Creating...

 Error: Provider produced inconsistent result after apply
 
 When applying changes to rhcs_cluster_rosa_classic.rosa_sts_cluster, provider "provider[\"registry.terraform.io/terraform-redhat/rhcs\"]" produced an unexpected new value: .proxy.additional_trust_bundle: was cty.StringVal("-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n"), but now cty.StringVal("REDACTED").

 This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

I'm using Terraform v1.5.3 with rhcs version 1.2.2.

**Comment by nirarg** [comment]
Hi @am90670 ,
Thank you for the feedback
We will check this and will update you as soon as possible

**Comment by nirarg** [comment]
Jira ticket: [OCM-3124](https://issues.redhat.com/browse/OCM-3124)

**Comment by am90670** [comment]
Is there an ETA on when this issue will be resolved?

**Comment by sagidayan** [comment]
Please put the jira ticket as a prefix to the PR title

**Comment by nirarg** [comment]
> Is there an ETA on when this issue will be resolved?

Hi @am90670 , no estimation on that yet

**Comment by cbell-work** [comment]
I have exactly the same issue, which is suspiciously similar to the case in #223 (failure when creating or updating a cluster `proxy.additional_trust_bundle` attribute), just no crash. I've attached my TF for validation.

Validation steps
1. Install Terraform CLI 1.5.2
1. Extract TF files into new, empty directory
1. Set auth vars for OCM and AWS SDKs

      ```bash
      export AWS_DEFAULT_REGION=us-east-1 AWS_ACCESS_KEY_ID='<SEKRET>' \
        AWS_SECRET_ACCESS_KEY='<SEKRET>' \
        TF_VAR_token='<SEKRET>'
      ```

1. Initialize Terraform

      ```bash
      rm -rf .terraform* terraform.tfstate*
      terraform init
      ```

1. Create cluster without trust bundle (note API will not accept proxy config without subnet IDs, hence the VPC creation TF code)

      ```bash
      unset TF_VAR_proxy_trust_bundle
      terraform apply
      ```

1. After cluster creation, add trust bundle and apply changes (I'm grabbing the first three CAs from the default trust bundle on Fedora here, but any source should do).

      ```bash
      export TF_VAR_proxy_trust_bundle="$(awk '$0 ~ /-----END CERTIFICATE-----/ { nc+=1 }; {if (nc >= 3) {print ; exit}}; {print}' /etc/pki/tls/certs/ca-bundle.crt)"
      terraform apply
      ```

Expected result: cluster-wide proxy configured as requested

Actual result:
```bash
rhcs_cluster_rosa_classic.rosa_sts_cluster: Modifying... [id=[rosa_test_vpc.tf.txt](https://github.com/terraform-redhat/terraform-provider-rhcs/files/12423845/rosa_test_vpc.tf.txt)]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to rhcs_cluster_rosa_classic.rosa_sts_cluster, provider "provider[\"registry.terraform.io/terraform-redhat/rhcs\"]" produced an unexpected new value: .proxy.additional_trust_bundle: was
│ cty.StringVal("<LONG_PEM_STRING_OF_3_CAS>"), but now cty.StringVal("REDACTED").
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```
Note that the "REDACTED" new value is from Terraform, not me. I did change the original value to "LONG_PEM_STRING_OF_3_CAS" for readability, however.

Terraform configs:
[rosa_test_vpc.tf](https://github.com/terraform-redhat/terraform-provider-rhcs/files/12423849/rosa_test_vpc.tf.txt)
[rosa_test.tf](https://github.com/terraform-redhat/terraform-provider-rhcs/files/12423846/rosa_test.tf.txt)

**Comment by sagidayan** [comment]
@cbell-work , @am90670 

It should be fixed in `v1.2.4`, published today.

Thanks for taking the time to submit this issue. I truly appreciate it 👏🏽
